From c202c264b0c0ffd8de44d223fde25961fa81ce9e Mon Sep 17 00:00:00 2001
From: yuiiio <atbjyk@protonmail.com>
Date: Thu, 24 Sep 2020 03:52:41 +0900
Subject: [PATCH 3/3] xfwm4 does not bypass compositor

---
 dlls/winex11.drv/window.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index bdd4582a234..52dac77c731 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -1051,6 +1051,13 @@ void update_net_wm_states( struct x11drv_win_data *data )
             XSendEvent( data->display, root_window, False,
                         SubstructureRedirectMask | SubstructureNotifyMask, &xev );
         }
+
+        // XFWM does not work _NET_WM_BYPASS_COMPOSITOR so dirty hack
+        if(bypass_compositor() && net_wm_bypass_compositor){
+            TRACE( "atbjyk's HACK: xfconf-query compositing off\n" );
+            char *compositor_off = "xfconf-query -c xfwm4 -p /general/use_compositing -s false";
+            system(compositor_off);
+        }
     }
     data->net_wm_state = new_state;
 
@@ -1193,6 +1200,13 @@ static void unmap_window( HWND hwnd )
             }
         }
 
+        // XFWM does not work _NET_WM_BYPASS_COMPOSITOR so dirty hack
+        if(bypass_compositor()){
+            TRACE( "atbjyk's HACK: xfconf-query compositing on\n" );
+            char *compositor_on = "xfconf-query -c xfwm4 -p /general/use_compositing -s true";
+            system(compositor_on);
+        }
+
         TRACE( "win %p/%lx\n", data->hwnd, data->whole_window );
 
         if (data->embedded) set_xembed_flags( data, 0 );
-- 
2.28.0

